securedrop/tests/functional/functional_test.py

```diff
@@ -162,55 +165,127 @@ def system(self, cmd):
         """
         Invoke a shell command. Primary replacement for os.system calls.
         """
-        print(cmd)
         ret = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
-                               stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+                               stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                               close_fds=True)
         out, err = ret.communicate()
         return out
 
     def setup(self, session_expiration=30):
 
-        signal.signal(signal.SIGUSR1, lambda _, s: traceback.print_stack(s))
-
+        self.localtesting = False
+        self.driver = None
+        self.second_driver = None
 
-        instance_information_path = join(FUNCTIONAL_TEST_DIR, 'instance_information.json')
+        instance_information_path = join(FUNCTIONAL_TEST_DIR,
+                                         'instance_information.json')
 
         if os.path.exists(instance_information_path):
             with open(instance_information_path) as fobj:
                 data = json.load(fobj)
-            self.source_location = data.get('source_location', "http://127.0.0.1:8080")
-            self.journalist_location = data.get('journalist_location', "http://127.0.0.1:8081")
+            self.source_location = data.get('source_location')
+            self.journalist_location = data.get('journalist_location')
             self.hidservauth = data.get('hidserv_token', '')
             self.admin_user = data.get('user')
             self.admin_user['totp'] = pyotp.TOTP(self.admin_user['secret'])
-            self.new_totp = None # To be created runtime
+            self.new_totp = None  # To be created runtime
             self.sleep_time = data.get('sleep_time', 10)
             if self.hidservauth:
                 if self.journalist_location.startswith('http://'):
                     location = self.journalist_location[7:]
                 self.add_hidservauth(location, self.hidservauth)
         else:
-            assert False, "Missing instance information JSON file."
+            self.localtesting = True
+            self.__context = journalist_app.create_app(config).app_context()
+            self.__context.push()
+
+            self.patcher2 = mock.patch('source_app.main.get_entropy_estimate')
+            self.mock_get_entropy_estimate = self.patcher2.start()
+            self.mock_get_entropy_estimate.return_value = 8192
+
+            env.create_directories()
+            self.gpg = env.init_gpg()
+            db.create_all()
+
+            # Add our test user
```

Comment:
Can we re-use `create-dev-data.py` for this?

Reply:
I tried, but, could not import from there, that is why added here.